### PR TITLE
Bug 1770

### PR DESCRIPTION
**Pull request by hippo91**
This PR fixes the bug #1770.
The bug was due to the fact that the test to determine if there are handlers able to deal with the exception was done on a generator and thus returns True even if there was no handlers.
Turning the generator into a list solves the problem.
Thanks @The-Compiler !

### CODE REVIEW

File: pylint/checkers/refactoring.py [hunk 1]
```diff
@@ -57,6 +57,7 @@ def _is_node_return_ended(node):
             return False
         exc_name = exc.pytype().split('.')[-1]
         handlers = utils.get_exception_handlers(node, exc_name)
+        handlers = list(handlers) if handlers is not None else list()
```
Comment (The-Compiler) [inline pylint/checkers/refactoring.py]: Not sure what pylint's coding style is here, but usually `[]` is more idiomatic than `list()`.

Comment (hippo91) [reply pylint/checkers/refactoring.py]: Ok! Thanks for the remark.

File: ChangeLog [hunk 2]
```diff
@@ -1,6 +1,13 @@
 ------------------
 Pylint's ChangeLog
 ------------------
+What's New in Pylint 2.0?
+=========================
+
+    * Fix a false positive ``inconsistent-return-statements`` message when if
+      statement is inside try/except.
+			Close #1770
```
Comment (PCManticore) [inline ChangeLog]: There are some wrong indents here

**Comment by PCManticore** [comment]
Also it would be nice if you backport this to 1.8 branch please. :)

**Comment by hippo91** [comment]
No problem for backporting to 1.8 branch but should i launch a special git command or just copy the files into 1.8 branch? And also should i add ChangeLog and Whatsnew entries or not?
Thanks

**Comment by PCManticore** [comment]
Yup, I do it with `git cherry-pick -n commit_hash`. The `-n` is for not having the changes automatically committed. You can also change the ChangeLog and News entries

**Comment by hippo91** [comment]
All right! Thank you @PCManticore ! I didn't know this command! 
Should i do a PR for 1.8 branch?

**Comment by PCManticore** [comment]
Feel free to push it straight in the branch :)

**Comment by hippo91** [comment]
All right! Let's do this! Thanks @PCManticore .

**Pull request merged**